File: Frontend/src/main/java/DataAccessObjects/CityDAO.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package DataAccessObjects;

import Entities.CityEntity;
import javax.ws.rs.client.Client;
import javax.ws.rs.client.ClientBuilder;
import javax.ws.rs.client.Entity;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

/**
 *
 * @author gonza
 */
public class CityDAO {

    Client client = ClientBuilder.newClient();

    // Select
    public CityEntity getCity(int id) {
        return client
                .target("http://localhost:8080/api/todo/")
                .path(String.valueOf(id))
                .request(MediaType.APPLICATION_JSON)
                .get(CityEntity.class);
    }

    //Create
    public Response createCity(CityEntity cityEntity) {
        return client
                .target("http://localhost:8080/api/todo")
                .request(MediaType.APPLICATION_JSON)
                .post(Entity.entity(cityEntity, MediaType.APPLICATION_JSON));
    }

    // Delete
    public CityEntity deletCity(int id) {
        return client
                .target("http://localhost:8080/api/todo/")
                .path(String.valueOf(id))
                .request(MediaType.APPLICATION_JSON)
                .delete(CityEntity.class);
    }

    // Update
    public Response updateCity(CityEntity cityEntity, int id) {
        return client
                .target("http://localhost:8080/api/todo")
                .path(String.valueOf(id))
                .request(MediaType.APPLICATION_JSON)
                .put(Entity.entity(cityEntity, MediaType.APPLICATION_JSON));
    }
}

File: FrontendVUE/touristran/src/DataAccessObjects/CityDAO.js
const axios = require("axios");

function getCities(callback) {
  axios
    .get("http://localhost:8080/api/cities/")
    .then(response => {
      //console.log(response);
      callback(response.data);
    })
    .catch(function(error) {
      console.log(error);
    });
}

function getCity(id, callback) {
  axios
    .get(`http://localhost:8080/api/cities/${id}`)
    .then(response => {
      //console.log(response);
      callback(response.data);
    })
    .catch(function(error) {
      console.log(error);
    });
}

function createCity(city, callback) {
  axios
    .post("http://localhost:8080/api/cities", city)
    .then(response => {
      //console.log(response);
      callback(response.status);
    })
    .catch(function(error) {
      console.log(error);
    });
}

function updateCity(city, callback) {
  axios
    .put("http://localhost:8080/api/cities/0", city)
    .then(response => {
      //console.log(response);
      callback(response.status);
    })
    .catch(function(error) {
      console.log(error);
    });
}

function deleteCity(id, callback) {
  axios
    .delete(`http://localhost:8080/api/cities/${id}`)
    .then(response => {
      //console.log(response);
      callback(response.status);
    })
    .catch(function(error) {
      console.log(error);
    });
}

export default {
  getCities,
  getCity,
  createCity,
  updateCity,
  deleteCity
};
